craft_application/models/platforms.py

```diff
@@ -38,19 +38,21 @@
 )
 
 
-def _validate_platform_name_not_reserved(name: str) -> str:
+def _validate_platform_name(name: str) -> str:
     """Validate that the given platform name is not reserved."""
     if name in RESERVED_PLATFORM_NAMES:
         raise ValueError(f"Reserved platform name: {name!r}")
+    if "/" in name:
+        raise ValueError("Platform names may not contain the '/' character.")
```

Comment:
```suggestion
        raise ValueError("Platform names cannot contain the '/' character.")
```

Most of the CLI exemplars use "cannot" extensively. I'm not sure about calling out characters, though. We could maybe reuse our documentation pattern, but it's wordier.